lib/utils/azure_queues.ts

```diff
@@ -35,77 +41,87 @@ export const getDelaySecForRetries = (
     .filter(nr => nr <= maxRetries)
     .map(nr => Math.ceil(minBackoff * Math.pow(2, nr) / 1000));
 
-export function queueMessageToString(context: IContext): string {
-  /* istanbul ignore next */
+/* istanbul ignore next */
+export function queueMessageToString(queueMessage: IQueueMessage): string {
   return [
-    "bindings = ",
-    JSON.stringify(context.bindings),
-    "; queueTrigger = ",
-    context.bindingData.queueTrigger,
+    "queueTrigger = ",
+    queueMessage.queueTrigger,
     "; expirationTime = ",
-    context.bindingData.expirationTime,
+    queueMessage.expirationTime,
     "; insertionTime = ",
-    context.bindingData.insertionTime,
+    queueMessage.insertionTime,
     "; nextVisibleTime = ",
-    context.bindingData.nextVisibleTime,
+    queueMessage.nextVisibleTime,
     "; id = ",
-    context.bindingData.id,
+    queueMessage.id,
     "; popReceipt = ",
-    context.bindingData.popReceipt,
+    queueMessage.popReceipt,
     "; dequeueCount = ",
-    context.bindingData.dequeueCount
+    queueMessage.dequeueCount
   ].join("");
 }
 
 /**
- * Schedule the re-processing of a message in the queue
- * after an incremental delay (visibilityTimeout).
+ * Update message visibilityTimeout with an incremental delay.
+ *
+ * You MUST call context.done(retryMsg) in the caller
+ * to schedule a retry (the re-processing of a message in the queue).
  *
  * Useful in case of transient errors. The message is enqueued with
  * a newly computed visibilityTimeout (proportional to dequeueCount)
  *
  * @param queueService  The Azure storage queue service
  * @param queueName     The Azure storage queue name
  * @param context       The Functions context with bindings
+ *
+ * @return              False if message is expired.
  */
-export function retryMessageEnqueue(
+export function updateMessageVisibilityTimeout<T extends IQueueMessage>(
   queueService: QueueService,
   queueName: string,
-  context: IContext
-): void {
-  const queueMessage = context.bindingData;
+  queueMessage: T
```

Comment:
I've refactored this function to not call context.done (just update the visibilityTimeout), which is a duty of the caller. In this way we can avoid to pass the whole context to this function.